### DISCUSSION TITLE
Debugger : Output flags/registers in trace files

### DISCUSSION THREAD

**Discussion by colinleroy** [mamedev/discussions#59] [Ideas]
Hi folks,
what would you think about dumping flags and register values in trace files? Going from
0485: ldy $2556
To
0485: ldy $2556    ; x=$0, y= $1f, z=0

This could tremendously help analyzing what happens, after it happened.
If you think this is a good idea, I'm willing to give it a shot (although I would maybe need a bit of guidance to get started).
Thanks!

**Comment by rb6502** [comment]
I think it's a good idea (I've wanted it myself on occasion), but we'd need some way to choose which registers should be included in the output.  A/X/Y/P/S is fine for a 6502 but dumping all 70+ registers for each line on a PowerPC 403 would be too much.

**Comment by cuavas** [comment]
Isn’t that exactly what the [tracelog](https://docs.mamedev.org/debugger/general.html#debugger-command-tracelog) and [tracesym](https://docs.mamedev.org/debugger/general.html#debugger-command-tracesym) commands are for?

There’s an example showing how to use `tracelog` in the examples for the [trace command](https://docs.mamedev.org/debugger/execution.html#debugger-command-trace).

**Comment by colinleroy** [comment]
Yes, in fact it's exactly that! 
I somehow missed it in the doc.
I love it when the thing is already done :)
